mmdet/models/dense_heads/embedding_rpn_head.py

```diff
@@ -0,0 +1,120 @@
+import mmcv
+import torch
+import torch.nn as nn
+from mmcv import tensor2imgs
+
+from mmdet.models.builder import HEADS
+from ...core import bbox_cxcywh_to_xyxy
+
+
+@HEADS.register_module()
+class EmbeddingRPNHead(nn.Module):
+    """RPNHead in the `Sparse R-CNN <https://arxiv.org/abs/2011.12450>`_ .
+
+    Unlike traditional RPNHead, this module does not need FPN input, but just
+    decode init_proposal_bboxes and expand first dimension of
+    init_proposal_bboxes and init_proposal_features to the batch_size.
+
+    Args:
+        num_proposals (int): Number of init_proposals. Default 100.
+        proposal_feature_channel (int): Channel number of
+            init_proposal_feature. Defaults to 256.
+    """
+
+    def __init__(self,
+                 num_proposals=100,
+                 proposal_feature_channel=256,
+                 **kwargs):
```

Comment:
Since this module does not inherit other customized modules, `**kwargs` might be unnecessary.

Reply:
The detector would pass `train_cfg` and `test_cfg` here.

@Johnson-Wang Johnson-Wang 10 days ago Collaborator
If it is always None, can we delete it?
@jshilong jshilong 10 days ago Author Contributor
Because this parameter must be passed in the top-level design, we can't delete it.
@Johnson-Wang Johnson-Wang 10 days ago Collaborator
Then how about use **kwargs?
@jshilong jshilong 10 days ago Author Contributor
sure, this is a better choice.